packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -279,6 +280,7 @@ const DataGrid = injectIntl((props) => {
     setFocusedCol(columnIndex);
 
     if (onColumnSelect) {
+      setLastActiveRowId('');
```

Comment:
The column header is a valid row as well. I would not expect us to have an empty value because we selected a column header.